### PR TITLE
chore: Add information on Security Considerations to constructs

### PR DESCRIPTION
**Pull request by ddneilson**
This is adding information on security considerations that customers should take into account when using the RFDK constructs. These are concrete recommendations, rather than security posture-type information.

Note: This is a repost of: https://github.com/aws/aws-rfdk/pull/80

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-rfdk/lib/deadline/lib/repository.ts [hunk 1]
```diff
@@ -347,18 +347,34 @@ export interface RepositoryProps {
...
+ *   bootstrap bucket, or enable AWS CloudTrail on your account to assist in post-incident analysis of compromised production
+ *   environments.
+ * - The file system that is created by, or provided to, this construct contains the data for Deadline's Repository file
+ *   system. This file system contains information about your submitted jobs, and the plugin scripts that are run by the
+ *   Deadline applications in your render farm. An actor that can modify the contents of this file system can cause your
+ *   Deadline applications to run code of their choosing. You should restrict access to this file system to only those who
+ *   require it.
+ * - The database that is created by, or provided to, this construct is used by Deadline to store data about its configuration,
+ *   submitted jobs, machine information and status, and so on. An actor with access to this database can read information
+ *   any information that is entered into Deadline, and modify the bevavior of your render farm. You should restrict access
```
Comment (jusiskin) [inline packages/aws-rfdk/lib/deadline/lib/repository.ts]: nit: redundant word

> ... can read ~~information~~ any information that is ...

File: packages/aws-rfdk/lib/core/lib/exporting-log-group.ts [hunk 2]
```diff
@@ -60,11 +60,20 @@ export interface ExportingLogGroupProps {
...
- * 1) The Lambda SingletonFunction that checks for the existence of the LogGroup;
- * 2) The CloudWatch LogGroup (if it didn't exist already);
- * 3) The CloudWatch Alarm watching log exportation failures;
- * 4) The CloudWatch Event Rule to schedule log exportation;
- * 5) The Lambda SingletonFunction, with role, to export log groups to S3 by schedule.
+ * - The Lambda SingletonFunction that checks for the existence of the LogGroup;
+ * - The CloudWatch LogGroup (if it didn't exist already);
+ * - The CloudWatch Alarm watching log exportation failures;
+ * - The CloudWatch Event Rule to schedule log exportation;
+ * - The Lambda SingletonFunction, with role, to export log groups to S3 by schedule.
```
Comment (jusiskin) [inline packages/aws-rfdk/lib/core/lib/exporting-log-group.ts]: minor style inconsistency nit: seems that we sometimes add semi-colons after each resource in the list and sometimes we use periods.

**Review by jusiskin — CHANGES_REQUESTED** [review]
Looks like you addressed all prior feedback from #80. Just two minor style things left.

File: packages/aws-rfdk/lib/core/lib/exporting-log-group.ts (hunk 2 above)
Comment (ddneilson) [reply packages/aws-rfdk/lib/core/lib/exporting-log-group.ts]: Technically, the semi-colons are proper, but periods seem way more common. I'll make it consistent.

File: packages/aws-rfdk/lib/core/lib/mongodb-instance.ts [hunk 3]
```diff
@@ -304,11 +304,27 @@ export interface IMongoDb extends IConnectable, IConstruct {
...
+ * Security Considerations
+ * ------------------------
+ * - The administrator credentials for MongoDB are stored in a Secret within AWS SecretsManager. You must strictly limit
+ *   access to this secret to only entities that require it.
+ * - The instances deployed by this construct download and run scripts from your CDK bootstrap bucket when that instance
+ *   is launched. You must limit write access to your CDK bootstrap bucket to prevent an attacker from modifying the actions
+ *   performed by these scripts. We strongly recommend that you either enable Amazon S3 server access logging on your CDK
+ *   bootstrap bucket, or enable AWS CloudTrail on your account to assist in post-incident analysis of compromised production
+ *   environments.
+ * - The EBS Volume that is created by, or provided to, this construct is used to store the contents of your MongoDB data. To
```
Comment (grbartel) [inline packages/aws-rfdk/lib/core/lib/mongodb-instance.ts:324]: Should we have a note about encryption at rest on this volume?  at least if they provide us one

File: packages/aws-rfdk/lib/deadline/lib/worker-fleet.ts [hunk 4]
```diff
@@ -279,10 +279,21 @@ abstract class WorkerInstanceFleetBase extends Construct implements IWorkerFleet
...
- * 3) A script asset which is uploaded to your deployment bucket used to configure the worker so it can connect to the Render Queue
- * 4) An aws-rfdk.CloudWatchAgent to configure sending logs to cloudwatch.
+ * - An EC2 Auto Scaling Group to maintain the number of instances.
+ * - An Instance Role and corresponding IAM Policy.
+ * - An Amazon CloudWatch log group that contains the Deadline Worker, Deadline Launcher, and instance-startup logs for the instances
+ *   in the fleet.
+ *
+ * Security Considerations
+ * ------------------------
+ * - The instances deployed by this construct download and run scripts from your CDK bootstrap bucket when that instance
```
Comment (grbartel) [inline packages/aws-rfdk/lib/deadline/lib/worker-fleet.ts:289]: We currently warn the user if they do not encrypt the worker fleets EBS volumes should we have notes about that here as well?

**Review by grbartel — COMMENTED** [review]
Looks good just a couple questions regarding EBS and encryption

File: packages/aws-rfdk/lib/deadline/lib/worker-fleet.ts (hunk 4 above)
Comment (ddneilson) [reply packages/aws-rfdk/lib/deadline/lib/worker-fleet.ts:289]: Yeah, that would make sense.

File: packages/aws-rfdk/lib/core/lib/mongodb-instance.ts (hunk 3 above)
Comment (ddneilson) [reply packages/aws-rfdk/lib/core/lib/mongodb-instance.ts:324]: Certainly doesn't hurt!

**Review by grbartel — APPROVED** [review]
Approved

**Review by jusiskin — APPROVED** [review]
Looks great! Thanks @ddneilson

**Pull request merged**